### PR TITLE
Replacing Persistence Solution for Instrumentation Events with Paper

### PR DESCRIPTION
**Pull request by bhariharan**
- Currently, a single instrumentation event is stored in a single file. As a result, each write creates a new file, and in order to check if max limit of events has been reached, we look up the number of files created.
- As the number of events increases, this approach isn't the best. What we need is a simple key-value store that is fast.
- [Paper](https://github.com/pilgr/Paper) is an open source library that offers this functionality.
- I replaced the existing implementation using `Paper` instead.
- <b>Good news:</b> no public APIs were broken as part of this exercise, so we aren't breaking `semver`.
- Only the internals of reads and writes have been updated.

### CODE REVIEW

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java [hunk 1]
```diff
@@ -303,28 +238,4 @@ private String encrypt(String data) {
     private String decrypt(String data) {
         return Encryptor.decrypt(data, encryptionKey);
     }
-
-    /**
-     * This class acts as a filter to identify only the relevant event files.
-     *
-     * @author bhariharan
-     */
-    private static class EventFileFilter implements FilenameFilter {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: We don't need this anymore - there's only one file.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java [hunk 2]
```diff
@@ -236,64 +224,11 @@ public int getMaxEvents() {
...
     private boolean shouldStoreEvent() {
-        final List<File> files = getAllFiles();
-        int fileCount = 0;
-        if (files != null) {
-            fileCount = files.size();
-        }
-        return (isLoggingEnabled && (fileCount < maxEvents));
-    }
-
-    private InstrumentationEvent fetchEvent(File file) {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: We don't need this anymore - fetching an event is straightforward.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java [hunk 3]
```diff
@@ -52,13 +50,12 @@
  */
 public class EventStoreManager {
 
+    private static final String FILENAME = "Event_Store";
```
Comment (brandonpage) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: Is first character capital and underscores the filename paradigm?

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java [hunk 4]
```diff
@@ -158,16 +156,9 @@ public InstrumentationEvent fetchEvent(String eventId) {
...
      */
-    public boolean deleteEvent(String eventId) {
-        if (TextUtils.isEmpty(eventId)) {
-            SalesforceAnalyticsLogger.e(context, TAG, "Invalid event ID supplied: " + eventId);
-            return false;
-        }
-        final String filename = eventId + filenameSuffix;
-        final File file = new File(rootDir, filename);
-        return file.delete();
+    public void deleteEvent(String eventId) {
```
Comment (brandonpage) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: SemVer?  You could `return !book.contains(eventId);` but I doubt anyone will ever care.  Probably not worth fixing.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java (hunk 2 above)
Comment (brandonpage) [reply libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: 🎉

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java (hunk 4 above)
Comment (bhariharan) [reply libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: Fixed in #2036

**Pull request merged**